Assets/Scripts/Construction/PlayerConstructionHandler.cs

```diff
@@ -49,10 +54,27 @@ private void ExecutePlacement()
 
             GameObject newSite = Instantiate(constructionSitePrefab, (Vector3)currentPlacementPosition, CurrentConstructionRotation);
             newSite.GetComponent<ConstructionSite>().Initialize(currentBuildingInfo);
+            // should be set based on building type
+            var requirements = new SerializableDictionary<ItemInfo, int>{
+                {ItemManager.Instance.Items.ElementAt(2), 5}, 
+                /*{ItemManager.Instance.Items.ElementAt(7), 3},
+                {ItemManager.Instance.Items.ElementAt(14), 8}*/
+                };
+            newSite.GetComponent<ConstructionSite>().SetRequiredResources(requirements);
 
             ExitPlacement();
         }
 
+        private void GiveResource() {
+            Ray looking = new Ray(cameraTransform.position, cameraTransform.forward);
+            bool isLooking = Physics.Raycast(looking, out RaycastHit hit, maxBuildDistance, constructionSiteCollisionMask);
+            
+            if (isLooking && hit.collider.gameObject.name == "Resource(Clone)") { //TEMP
+                hit.collider.gameObject.GetComponentInParent<ConstructionSite>().DecreaseResourceRequirement(ItemManager.Instance.Items.ElementAt(2));
+                hit.collider.gameObject.GetComponentInParent<ConstructionSite>().IsReadyToBuild(finishedBuildingPrefab);
+            }
+        }
+
```

Comment:
as @Acu1000 said, but we don't have player/robot with ability to hold items now, so just move that InputSystem logic to new script something like "DebugConstructionItemProvider" or something

Reply:
I have already reworked InputSystem so it accepts physical items being dropped onto construction site using player's gravity gun. Should I stick with previous InputSystem or will the current version suffice?

Reply:
If it is an independent script, then ok. For now we just need to mock this functionality so it's implementation doesn't matter. It will be replaced when we implement robots or player EQ